### PR TITLE
dev-python/python-lsp-server: enable py3.11

### PR DESCRIPTION
**Pull request by peeweep**

### CODE REVIEW

**Comment by thesamesam** [comment]
One package per commit but ideally please use a valid signoff.

**Comment by thesamesam** [comment]
Are you running the test suites for these, by the way?

**Comment by peeweep** [comment]
@thesamesam There are many dependencies controlled by use flags that I have not checked locally, and I need to ask @gentoo-repo-qa-bot to check for me. Is there a way for me to run qa locally?

**Comment by thesamesam** [comment]
> @thesamesam There are many dependencies controlled by use flags that I have not checked locally, and I need to ask @gentoo-repo-qa-bot to check for me. Is there a way for me to run qa locally?

You can run 'pkgcheck scan' locally and/or 'pkgcheck scan --commits' which will check everything you did at least, although it wouldn't check reverse dependencies, that doesn't matter for this.

But make sure you are running the test suite for any package you add Python 3.11 to:
emerge -av1o --with-test-deps foo
FEATURES=test emerge -v1 foo

**Comment by AndrewAmmerlaan** [comment]
Last time I checked the test suite of jedi didn't pass for me with python3_11, does it work now?

File: dev-python/nose/nose-1.3.7_p20211111_p1-r1.ebuild [hunk 1]
```diff
@@ -4,7 +4,7 @@
 EAPI=8
 
 DISTUTILS_USE_PEP517=setuptools
-PYTHON_COMPAT=( python3_{8..10} pypy3 )
+PYTHON_COMPAT=( python3_{8..11} pypy3 )
```
Comment (AndrewAmmerlaan) [inline dev-python/nose/nose-1.3.7_p20211111_p1-r1.ebuild:7]: Nose is a package that we don't want to port to python3_11, it doesn't work with python3_11 and is not actively maintained upstream any more. Instead ebuilds that depend on it to run the tests should be ported to pytest, often this is an easy change: `distutils_enable_tests nose` --> `distutils_enable_tests pytest`. Sometimes it is more complicated though.

**Review by AndrewAmmerlaan — CHANGES_REQUESTED** [review]

File: dev-python/nose/nose-1.3.7_p20211111_p1-r1.ebuild (hunk 1 above)
Comment (arthurzam) [reply dev-python/nose/nose-1.3.7_p20211111_p1-r1.ebuild:7]: As the current maintainer of nose fork we use at gentoo, I can say that nose **is not** ready for python 3.11!

Comment (peeweep) [reply dev-python/nose/nose-1.3.7_p20211111_p1-r1.ebuild:7]: Thanks, nose is dev-python/astroid's test bdepend, I will try to use pytest instead of nose.

**Comment by peeweep** [comment]
It looks like there are too many software package tests to be modified. I will slowly update the packages that has passed the test.

**Pull request closed without merging**